conjure-go-client/httpclient/metrics.go

```diff
@@ -222,3 +231,25 @@ func (m *metricsWrappedConn) Close() error {
 	m.counter.Dec(1)
 	return m.Conn.Close()
 }
+
+func isTimeoutError(respErr error) bool {
+	if respErr == nil {
+		return false
+	}
+	rootErr := werror.RootCause(respErr)
+	if rootErr == nil {
+		return false
+	}
+
+	if nerr, ok := rootErr.(net.Error); ok && nerr.Timeout() {
+		return true
+	}
+	if rootErr == context.Canceled || rootErr == context.DeadlineExceeded {
```

Comment:
lets use `errors.Is(rootErr, context.Canceled) || ...` instead of a direct comparison